bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -600,12 +600,16 @@ public boolean isAborted() {
 
   @Override
   public void truncateTable(TableName tableName, boolean preserveSplits) throws IOException {
-    throw new UnsupportedOperationException("truncateTable");  // TODO
+    HTableDescriptor tableDescriptor = getTableDescriptor(tableName);
+    LOG.info("Deleting table %s", tableName);
+    deleteTable(tableName);
+    LOG.info("Creating table %s", tableName);
+    createTable(tableDescriptor);
   }
 
   @Override
   public boolean isTableAvailable(TableName tableName, byte[][] splitKeys) throws IOException {
```

Comment:
In terms of Bigtable, nothing.  In terms of HBase, tableExists() checks for the existence of the table.  The isTableAvailable() method checks if the hbase servers are up and there is an hbase regionServer that contains the table.